src/__tests__/message-id.integration.test.js

```diff
@@ -0,0 +1,273 @@
+import { modelName as messageFragmentModelName } from '../models/message-fragment';
+import { transferOutEhrCore } from '../services/transfer/transfer-out-ehr-core';
+import {
+  extractEbXmlData,
+  extractReferencedFragmentMessageIds
+} from '../services/parser/extract-eb-xml-data';
+import { transferOutFragments } from '../services/transfer/transfer-out-fragments';
+import { getNewMessageIdByOldMessageId } from '../services/database/message-id-replacement-repository';
+
+describe('Replacement of message IDs', () => {
+  // ============ COMMON PROPERTIES ============
+  const gp2gpUrl = 'http://fake-gp2gpmessager-url';
+  const gp2gpAuth = 'gp2gp-auth';
+  const gp2gpHeaders = { reqheaders: { authorization: auth => auth === gp2gpAuth } };
+  const ehrRepoUrl = 'http://fake-ehr-repo-url';
+  const ehrRepoAuth = 'ehr-repo-auth';
+  const ehrRepoHeaders = { reqheaders: { authorization: auth => auth === ehrRepoAuth } };
+
+  const conversationId = uuidv4();
+  const nhsNumber = '1234567890';
+  const odsCode = 'fake-ods-code';
+  const ehrRequestId = uuidv4();
+
+  const coreMessageId = 'DF91D420-DDC7-11ED-808B-AC162D1F16F0';
+  const fragmentMessageIds = [
+    'DFBA6AC0-DDC7-11ED-808B-AC162D1F16F0',
+    'DFEC7740-DDC7-11ED-808B-AC162D1F16F0',
+    'DFEC7741-DDC7-11ED-808B-AC162D1F16F0',
+    'DFF61430-DDC7-11ED-808B-AC162D1F16F0'
+  ];
+  const conversationIdFromEhrIn = '5037ae3a-0ce2-4f47-adb5-8fdd92e6f5a4';
+  const ehrCorePresignedUrl = 'http://fake-core-presigned-url';
+
+  const MessageIdReplacement = ModelFactory.getByName(messageIdReplacementModelName);
+  const RegistrationRequest = ModelFactory.getByName(registrationRequestModelName);
+  const MessageFragment = ModelFactory.getByName(messageFragmentModelName);
+  // ================= END =================
+
+  // ================= SETUP AND TEARDOWN =================
+  beforeAll(async () => {
+    logger.add(transportSpy);
+
+    process.env.GP2GP_MESSENGER_SERVICE_URL = gp2gpUrl;
+    process.env.GP2GP_MESSENGER_AUTHORIZATION_KEYS = gp2gpAuth;
+    process.env.EHR_REPO_SERVICE_URL = ehrRepoUrl;
+    process.env.EHR_REPO_AUTHORIZATION_KEYS = ehrRepoAuth;
+
+    // clear all records in database before test start
+    await RegistrationRequest.destroy({ where: {}, force: true });
+    await MessageFragment.destroy({ where: {}, force: true });
+    await MessageIdReplacement.destroy({ where: {}, force: true });
+  });
+
+  afterAll(async () => {
+    delete process.env.GP2GP_MESSENGER_SERVICE_URL;
+    delete process.env.GP2GP_MESSENGER_AUTHORIZATION_KEYS;
+    delete process.env.EHR_REPO_SERVICE_URL;
+    delete process.env.EHR_REPO_AUTHORIZATION_KEYS;
+
+    // close the db connection to avoid "Jest did not exit" warning messages
+    await ModelFactory.sequelize.close();
+  });
+  // ================= END SETUP AND TEARDOWN =================
+
+  // ================= HELPER FUNCTIONS =================
```

Comment:
Would it be beneficial to extract these helper functions out into a separate utility script?